huaweicloud/services/evs/data_source_huaweicloud_evs_volumes.go

```diff
@@ -0,0 +1,231 @@
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"service_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"shareable": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"create_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"update_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tags": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"wwn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func buildQueryOpts(d *schema.ResourceData) cloudvolumes.ListOpts {
+	return cloudvolumes.ListOpts{
+		AvailabilityZone:     d.Get("availability_zone").(string),
+		DedicatedStorageID:   d.Get("dedicated_storage_id").(string),
+		DedicatedStorageName: d.Get("dedicated_storage_name").(string),
+		EnterpriseProjectID:  d.Get("enterprise_project_id").(string),
+		Multiattach:          d.Get("shareable").(bool),
+		ServerID:             d.Get("server_id").(string),
+		Status:               d.Get("status").(string),
+	}
+}
+
+func sourceEvsAttachment(attachements []cloudvolumes.Attachment, mode string) []map[string]interface{} {
+	result := make([]map[string]interface{}, len(attachements))
+	for i, attachement := range attachements {
+		result[i] = map[string]interface{}{
+			"id":            attachement.AttachmentID,
+			"attached_at":   attachement.AttachedAt,
+			"attached_mode": mode,
+			"device":        attachement.Device,
+			"server_id":     attachement.ServerID,
+		}
+	}
+	return result
+}
+
+func sourceEvsVolumes(volumes []cloudvolumes.Volume) ([]map[string]interface{}, []string) {
+	result := make([]map[string]interface{}, len(volumes))
+	ids := make([]string, len(volumes))
+
+	for i, volume := range volumes {
+		vMap := map[string]interface{}{
+			"id":                    volume.ID,
+			"attachments":           sourceEvsAttachment(volume.Attachments, volume.Metadata.AttachedMode),
+			"availability_zone":     volume.AvailabilityZone,
+			"description":           volume.Description,
+			"enterprise_project_id": volume.EnterpriseProjectID,
+			"shareable":             volume.Multiattach,
+			"name":                  volume.Name,
+			"service_type":          volume.ServiceType,
+			"size":                  volume.Size,
+			"status":                volume.Status,
+			"create_at":             volume.CreatedAt,
+			"update_at":             volume.UpdatedAt,
+			"tags":                  volume.Tags,
+		}
+		bootable, err := strconv.ParseBool(volume.Bootable)
+		if err != nil {
+			logp.Printf("[ERROR] The bootable connot be converted from boolen format to string format.")
```

Comment:
Errors must be handled, it is recommended to return this error to the console

Reply:
The error message has been handled.